library/lua/class.lua

```diff
@@ -15,6 +15,10 @@ reserved_names = { super = true, ATTRS = true }
 attrs_meta = attrs_meta or {}
 
 -- Create or updates a class; a class has metamethods and thus own metatable.
+---@generic T: table
+---@param class? T
+---@param parent? table
+---@return table|T
```

Comment:
is the return type `table|T` because `class` can be nil?

Reply:
Exactly right if class is nill we return a new table. Technically we should define a new class type above this that includes common methods and IT should be the return type but baby steps!